pennylane/math/single_dispatch.py

```diff
@@ -0,0 +1,321 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Autoray registrations"""
+# pylint:disable=protected-access,import-outside-toplevel,wrong-import-position
+from importlib import import_module
+import numbers
+
+import autoray as ar
+import numpy as np
+
+
+def _i(name):
+    """Convenience function to import PennyLane
+    interfaces via a string pattern"""
+    if name == "tf":
+        return import_module("tensorflow")
+
+    if name == "qml":
+        return import_module("pennylane")
+
+    return import_module(name)

```

Comment:
This is added to make `import_module("tensorflow")` into `_i('tf')` right? This is just to make the lambdas below shorter?

Reply:
Yes, what you are seeing here is the result of very long battle between me and black.